java/acorn-comparison/src/main/java/com/datasqrl/ai/comparison/trace/TraceContext.java

```diff
@@ -0,0 +1,55 @@
+package com.datasqrl.ai.comparison.trace;
+
+import com.datasqrl.ai.tool.Context;
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public class TraceContext implements Context {
+
+  private int requestId;
+  private int invocationId;
+  private final Map<String, Object> secure;
+
+  public static TraceContext of(Map<String, Object> secure) {
+    return new TraceContext(0, 0, secure);
+  }
+
+  public static TraceContext convert(Context context) {
+    Preconditions.checkArgument(context instanceof TraceContext, "Expected a TraceContext");
+    return (TraceContext) context;
+  }
```

Comment:
@mbroecheler I can't make sense of this code. I am trying to convert a regular context into a TraceContext but it will always fail with the precondition

Reply:
Try creating a TraceContext instead of a normal ContextImpl.

Reply:
But in which situation will this code work? why would I call TraceContext.convert(context), when context is already a TraceContext? 🤷

Reply:
It's only used in the TraceRecorder and TraceChatProvider to convert a generic Context back to a TraceContext. It's essentially a local helper method. We could make it package-level visibility.